### ISSUE TITLE
[Labs/Select] Errors in popper.js typings 

### ISSUE THREAD

**Issue by slavasitnikov** [palantir/blueprint#1527]
### Bug report

- __Package version(s)__:  typescript: 2.5.2, blueprintjs/labs: 0.8.0

#### Steps to reproduce

https://gist.github.com/slavasitnikov/887e240be9b08b35d0b67516d4e6ce08

```
npm i
./node_modules/.bin/tsc
```

#### Actual behavior
Compilation errors:

```
node_modules/popper.js/index.d.ts(38,5): error TS2411: Property 'shift' of type 'BaseModifier | undefined' is not assignable to string index type 'BaseModifier & Record<string, any>'.
node_modules/popper.js/index.d.ts(39,5): error TS2411: Property 'offset' of type '(BaseModifier & { offset?: string | number | undefined; }) | undefined' is not assignable to string index type 'BaseModifier & Record<string, any>'.
node_modules/popper.js/index.d.ts(42,5): error TS2411: Property 'preventOverflow' of type '(BaseModifier & { priority?: Position[] | undefined; padding?: number | undefined; boundariesElem...' is not assignable to string index type 'Base
Modifier & Record<string, any>'.
node_modules/popper.js/index.d.ts(47,5): error TS2411: Property 'keepTogether' of type 'BaseModifier | undefined' is not assignable to string index type 'BaseModifier & Record<string, any>'.
node_modules/popper.js/index.d.ts(48,5): error TS2411: Property 'arrow' of type '(BaseModifier & { element?: string | Element | undefined; }) | undefined' is not assignable to string index type 'BaseModifier & Record<string, any>'.
node_modules/popper.js/index.d.ts(51,5): error TS2411: Property 'flip' of type '(BaseModifier & { behavior?: Position[] | "flip" | "clockwise" | "counterclockwise" | undefined; ...' is not assignable to string index type 'BaseModifier &
Record<string, any>'.
node_modules/popper.js/index.d.ts(56,5): error TS2411: Property 'inner' of type 'BaseModifier | undefined' is not assignable to string index type 'BaseModifier & Record<string, any>'.
node_modules/popper.js/index.d.ts(57,5): error TS2411: Property 'hide' of type 'BaseModifier | undefined' is not assignable to string index type 'BaseModifier & Record<string, any>'.
node_modules/popper.js/index.d.ts(58,5): error TS2411: Property 'applyStyle' of type '(BaseModifier & { onLoad?: Function | undefined; gpuAcceleration?: boolean | undefined; }) | unde...' is not assignable to string index type 'BaseModif
ier & Record<string, any>'.
```

It seems it is the same bug here: https://github.com/FezVrasta/popper.js/issues/414

**Comment by DrewDennison** [comment]
I fixed (for now) by removing strict null checks from TS and got it to compile in case that helps anyone.

**Comment by jtheisen** [comment]
This is fixed in popper.js 1.12.5 and greater.

It's [this issue](https://github.com/FezVrasta/popper.js/pull/367).

**Comment by jtheisen** [comment]
As a temporary fix, update [this line](https://github.com/jtheisen/popper.js/blob/b762974f5378c68686678106fa2ccbc806048f9d/packages/popper/index.d.ts#L68) in your `index.d.ts`.

**Comment by thelastnode** [comment]
Looks like this is [fixed in v1.11.1](https://github.com/jtheisen/popper.js/blob/v1.11.1/packages/popper/index.d.ts#L68) as well. That might be an easier version bump to get out a quick fix?

**Comment by mgambati** [comment]
Can this be fixed on next labs 0.12.0 release?

**Comment by adidahiya** [comment]
Hey, I figured this out. webpack 3 defaults to loading the "module" entry from package.json, which is a breaking change since webpack 1. I got this working with the latest popper.js in my PR here: https://github.com/palantir/blueprint/pull/1786/commits/4c141b92d1b3e765f99f8dd083232d63bdf2ebeb

Should be shipped in the next release.
